crates/cdk-common/src/database/wallet.rs

```diff
@@ -118,4 +118,23 @@ pub trait Database: Debug {
     ) -> Result<Vec<Transaction>, Self::Err>;
     /// Remove transaction from storage
     async fn remove_transaction(&self, transaction_id: TransactionId) -> Result<(), Self::Err>;
+
+    // --- P2PK signing key storage ---
+    /// Store a P2PK signing key. Implementations should upsert by derived pubkey.
+    /// Default no-op for backends that don't support it.
+    async fn add_p2pk_key(&self, _secret_key: SecretKey) -> Result<(), Self::Err> {
+        Ok(())
+    }
+    /// Get a stored P2PK secret key by pubkey. Default returns None.
+    async fn get_p2pk_key(&self, _pubkey: PublicKey) -> Result<Option<SecretKey>, Self::Err> {
+        Ok(None)
+    }
+    /// List all stored P2PK signing keys. Default returns empty list.
+    async fn list_p2pk_keys(&self) -> Result<Vec<SecretKey>, Self::Err> {
+        Ok(vec![])
+    }
+    /// Remove a stored P2PK signing key by pubkey. Default no-op.
+    async fn remove_p2pk_key(&self, _pubkey: PublicKey) -> Result<(), Self::Err> {
+        Ok(())
```

Comment:
I don't think these should be no ops especially ones that do not return errors we should force db backends to support this.

Reply:
Updated the trait so these methods are mandatory (no default no-ops) and wired them through every wallet backend. The SQL store keeps its existing table, Redb has a v5 migration to create p2pk_signing_keys, and the FFI bridge now exposes the concrete impls, so all backends persist and surface P2PK keys consistently.